### ISSUE TITLE
WaveNets use insane amount of CPU 

### ISSUE THREAD

**Issue by 38github** [mikeoliphant/neural-amp-modeler-lv2#79]
I use Clang 18 and have compiled (master) with:
* `cmake .. -DCMAKE_BUILD_TYPE="Release" -DRTNEURAL_EIGEN=ON -DLSTM_PREFER_NAM=OFF -DWAVENET_PREFER_NAM=OFF -DRTNEURAL_USE_AVX=ON`
* `cmake .. -DCMAKE_BUILD_TYPE="Release" -DRTNEURAL_EIGEN=ON -DLSTM_PREFER_NAM=OFF -DWAVENET_PREFER_NAM=ON -DRTNEURAL_USE_AVX=ON`
* `cmake .. -DCMAKE_BUILD_TYPE="Release" -DRTNEURAL_EIGEN=ON -DLSTM_PREFER_NAM=OFF -DWAVENET_PREFER_NAM=ON -DRTNEURAL_USE_AVX=OFF`

They (WaveNets) all use insane amount of CPU. Using NAM for WaveNet hasn't solved it for me.

**Comment by mikeoliphant** [comment]
What architectuer/OS? Have you tried the pre-built binaries from the release?

**Comment by 38github** [comment]
Message ID: ***@***.***>It doesn't happen with the pre-built v3 using OpenSUSE Tumbleweed (x86_64). I will do an aarch64 build soon and see if my builds behave the same on that architecture. I use Clang-18 and haven't seen this behavior before.

I will get back to you!

**Comment by mikeoliphant** [comment]
Looks like you are just missing "-DUSE_NATIVE_ARCH=ON"

**Comment by 38github** [comment]
> Looks like you are just missing "-DUSE_NATIVE_ARCH=ON"

Yes but I am doubtful that not using it versus using it would go from 40% 1% CPU. I think that I need to look over my compiler flags that gave no issues in earlier versions.

**Comment by mikeoliphant** [comment]
Have you figured this out yet?

**Comment by 38github** [comment]
It seems that -fopenmp might screw something up. When I compiled it again with Clang but only used your -DUSE_NATIVE_ARCH=ON I do not encounter the excess CPU usage. From now on if I do experience issues I will first compile it with the default flags so that we can rule out that at least.

**Comment by mikeoliphant** [comment]
👍
